File: robert-server-ws-rest/src/test/java/fr/gouv/stopc/robertserver/ws/test/RestAssuredManager.java
package fr.gouv.stopc.robertserver.ws.test;

import fr.gouv.stopc.robertserver.common.RobertClock;
import io.restassured.RestAssured;
import org.hamcrest.Matcher;
import org.springframework.test.context.TestContext;
import org.springframework.test.context.TestExecutionListener;

import static org.hamcrest.Matchers.equalTo;

public class RestAssuredManager implements TestExecutionListener {

    private static RobertClock clock;

    @Override
    public void beforeTestMethod(TestContext testContext) {
        RestAssured.port = testContext.getApplicationContext()
                .getEnvironment()
                .getRequiredProperty("local.server.port", Integer.class);
        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
        clock = testContext.getApplicationContext().getBean(RobertClock.class);
    }

    public static Matcher<Long> equalToServiceStartNtpTimeStamp() {
        return equalTo(clock.atEpoch(0).asNtpTimestamp());
    }
}

File: robert-server-ws-rest/src/test/kotlin/fr/gouv/stopc/robertserver/ws/test/RestAssuredManager.kt
package fr.gouv.stopc.robertserver.ws.test

import fr.gouv.stopc.robertserver.common.RobertClock
import io.restassured.RestAssured
import io.restassured.specification.RequestSpecification
import org.springframework.test.context.TestContext
import org.springframework.test.context.TestExecutionListener

fun When() = RestAssured.`when`()
fun RequestSpecification.When() = this.`when`()

object RestAssuredManager : TestExecutionListener {

    private lateinit var clock: RobertClock

    override fun beforeTestMethod(testContext: TestContext) {
        RestAssured.port = testContext.applicationContext
            .environment
            .getRequiredProperty("local.server.port", Int::class.java)
        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails()
        clock = testContext.applicationContext.getBean(RobertClock::class.java)
    }
}
